include/deal.II/non_matching/mapping_info.h

```diff
@@ -646,288 +1065,362 @@ namespace NonMatching
-        ExcMessage(
-          "cell_index has to be specified if face_number is specified!"));
   }
 
 
 
-  template <int dim, int spacedim>
-  const Mapping<dim, spacedim> &
-  MappingInfo<dim, spacedim>::get_mapping() const
+  template <int dim, int spacedim, typename Number>
+  unsigned int
+  MappingInfo<dim, spacedim, Number>::calculate_compressed_cell_index(
+    const unsigned int cell_index) const
   {
-    return *mapping;
+    if (do_cell_index_compression)
+      {
+        Assert(cell_index_to_compressed_cell_index[cell_index] !=
+                 numbers::invalid_unsigned_int,
+               ExcMessage("Mapping info object was not initialized for this"
+                          " active cell index!"));
+        return cell_index_to_compressed_cell_index[cell_index];
+      }
+    else
+      return cell_index;
   }
 
 
+  template <int dim, int spacedim, typename Number>
+  void
+  MappingInfo<dim, spacedim, Number>::store_unit_points(
+    unsigned int                   unit_points_index_offset,
+    const unsigned int             n_q_points,
+    const unsigned int             n_q_points_unvectorized,
+    const std::vector<Point<dim>> &points)
+  {
+    const unsigned int n_lanes =
+      dealii::internal::VectorizedArrayTrait<VectorizedArrayType>::width;
 
-  template <int dim, int spacedim>
-  UpdateFlags
-  MappingInfo<dim, spacedim>::get_update_flags() const
+    for (unsigned int q = 0; q < n_q_points; ++q)
+      {
+        const unsigned int offset = unit_points_index_offset + q;
+        for (unsigned int v = 0;
+             v < n_lanes && q * n_lanes + v < n_q_points_unvectorized;
+             ++v)
+          for (unsigned int d = 0; d < dim; ++d)
+            dealii::internal::VectorizedArrayTrait<VectorizedArrayType>::get(
+              unit_points[offset][d], v) = points[q * n_lanes + v][d];
+      }
+  }
+
+
+
+  template <int dim, int spacedim, typename Number>
+  void
+  MappingInfo<dim, spacedim, Number>::store_mapping_data(
+    unsigned int                    unit_points_index_offset,
+    const unsigned int              n_q_points,
+    const unsigned int              n_q_points_unvectorized,
+    const MappingInfo::MappingData &mapping_data)
   {
-    return update_flags;
+    const unsigned int n_lanes =
+      dealii::internal::VectorizedArrayTrait<Number>::width;
+
+    for (unsigned int q = 0; q < n_q_points; ++q)
+      {
+        const unsigned int offset = unit_points_index_offset + q;
+        for (unsigned int v = 0;
+             v < n_lanes && q * n_lanes + v < n_q_points_unvectorized;
+             ++v)
+          {
+            if (update_flags_mapping & UpdateFlags::update_jacobians)
+              for (unsigned int d = 0; d < dim; ++d)
+                for (unsigned int s = 0; s < spacedim; ++s)
+                  dealii::internal::VectorizedArrayTrait<Number>::get(
+                    jacobians[offset][d][s], v) =
+                    mapping_data.jacobians[q * n_lanes + v][d][s];
```

Comment:
Have you considered using
https://github.com/dealii/dealii/blob/4ec36db9eed1e46fb96b5787af7e80d9ab45b180/include/deal.II/fe/mapping_q_internal.h#L1242-L1272
I agree it is a bit more complicated because we might or might not vectorize, but it feels that function would fit.

Reply:
It is the other way around here because we store something scalar in a (potentially) vectorized destination.